api/app/routers/fba.py

```diff
@@ -33,31 +34,64 @@ async def get_all_fire_centers(_=Depends(authentication_required)):
     return FireCenterListResponse(fire_centers=fire_centers)
 
 
-@router.get('/fire-zone-areas/{run_type}/{run_date}/{for_date}',
+@router.get('/fire-zone-areas/{run_type}/{run_datetime}/{for_date}',
             response_model=FireZoneAreaListResponse)
-async def get_zones(run_type: RunType, run_date: date, for_date: date, _=Depends(authentication_required)):
+async def get_zones(run_type: RunType, run_datetime: datetime, for_date: date, _=Depends(authentication_required)):
     """ Return area of each zone, and percentage of area of zone with high hfi. """
     async with get_async_read_session_scope() as session:
         zones = []
 
         rows = await get_hfi_area(session,
                                   RunTypeEnum(run_type.value),
-                                  run_date,
+                                  run_datetime,
                                   for_date)
 
         # Fetch rows.
         for row in rows:
-            combustible_area = row.combustible_area
-            hfi_area = row.hfi_area
-
+            combustible_area = row.combustible_area  # type: ignore
+            hfi_area = row.hfi_area  # type: ignore
             zones.append(FireZoneArea(
-                mof_fire_zone_id=row.source_identifier,
-                elevated_hfi_area=row.hfi_area,
+                mof_fire_zone_id=row.source_identifier,  # type: ignore
+                elevated_hfi_area=row.hfi_area,  # type: ignore
                 elevated_hfi_percentage=hfi_area / combustible_area * 100))
         return FireZoneAreaListResponse(zones=zones)
 
 
-@router.get('/sfms_run_datetimes/{run_type}/{for_date}', response_model=List[datetime])
+@router.get('/hfi-fuels/{run_type}/{for_date}/{run_datetime}', response_model=List[FireZoneHfiThresholdsByFuelType])
+async def get_hfi_thresholds_by_fuel_type(run_type: RunType,
+                                          for_date: date,
+                                          run_datetime: datetime):
+    """
+    Get the fuel types for the run_type, for_date, run_date
+    """
+    logger.info('hfi-fuels/%s/%s/%s', run_type.value, for_date, run_datetime)
+    async with get_async_read_session_scope() as session:
+        fuel_types_high_hfi = await get_fuel_types_with_high_hfi(
+            session,
+            run_type=RunTypeEnum(run_type.value),
+            for_date=for_date,
+            run_datetime=run_datetime
+        )
+        fire_zones_hfi_fuel_types = []
+        current_zone_id = None
+        for row in fuel_types_high_hfi:
```

Comment:
What is this loop doing?

Reply:
iterating through the list of raw SQL row results and building out a response in the format of `List[FireZoneHfiThresholdsByFuelTypes]`

Reply:
example from my local:
```
{
  "mof_fire_zone_id": 401,
  "fuel_types": [
    {
      "fuel_type_id": 2,
      "threshold": 2,
      "area": 413180.88182108715
    },
    {
      "fuel_type_id": 3,
      "threshold": 1,
      "area": 16036589.21019188
    },
    {
      "fuel_type_id": 3,
      "threshold": 2,
      "area": 0
    },
    {
      "fuel_type_id": 4,
      "threshold": 1,
      "area": 13854687.478280807
    },
    {
      "fuel_type_id": 5,
      "threshold": 1,
      "area": 0
    },
    {
      "fuel_type_id": 5,
      "threshold": 2,
      "area": 0
    },
    {
      "fuel_type_id": 6,
      "threshold": 1,
      "area": 0.000003965822036318498
    },
    {
      "fuel_type_id": 7,
      "threshold": 1,
      "area": 64173011.49065131
    },
    {
      "fuel_type_id": 7,
      "threshold": 2,
      "area": 172.81822308288335
    },
    {
      "fuel_type_id": 8,
      "threshold": 1,
      "area": 8.85075916702494
    },
    {
      "fuel_type_id": 8,
      "threshold": 2,
      "area": 0
    },
    {
      "fuel_type_id": 9,
      "threshold": 1,
      "area": 0
    },
    {
      "fuel_type_id": 12,
      "threshold": 1,
      "area": 173.70439270479034
    },
    {
      "fuel_type_id": 12,
      "threshold": 2,
      "area": 171.4817892202029
    },
    {
      "fuel_type_id": 14,
      "threshold": 1,
      "area": 264783760.6426373
    },
    {
      "fuel_type_id": 14,
      "threshold": 2,
      "area": 80245661.31049424
    },
    {
      "fuel_type_id": 99,
      "threshold": 1,
      "area": 1.212138028838794e-16
    },
    {
      "fuel_type_id": 99,
      "threshold": 2,
      "area": 0
    },
    {
      "fuel_type_id": 102,
      "threshold": 1,
      "area": 0
    }
  ]
}
```

Reply:
Makes me nervous how many conditionals and state mutations are in there, why not just:
```
        fuel_stats_by_fire_zone = groupby(fuel_types_high_hfi, operator.itemgetter(0))
        fire_zone_stats = dict((k, list(map(lambda x: x, values)))
                               for k, values in fuel_stats_by_fire_zone)
```
Then iterate over the existing dictionary of `fire_zone_id` ->  (`fuel_type_id`, `threshold`, `area`)

Reply:
because I prefer to live and code dangerously! 😈

jk, I'll make that change now

Reply:
Here it is: https://github.com/bcgov/wps/compare/story/asa-fuel-layer-data/2208...task/asa-fuel-layer-data-endpoint?expand=1

Reply:
@conbrad  what's that link for? When I click it, I'm getting a prompt to open a new PR

Reply:
FYI I just pushed a change to implement your feedback

Reply:
> @conbrad what's that link for? When I click it, I'm getting a prompt to open a new PR

Scroll down to see the diff that implements a fully working version of what I was talking about

Reply:
But yeah looks like you got it, nice!